app/src/main/java/com/everest/movieapp/ui/CurrentYearMovies.kt

```diff
@@ -0,0 +1,60 @@
+package com.everest.movieapp.ui
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.recyclerview.widget.RecyclerView
+import com.everest.movieapp.R
+import com.everest.movieapp.adapters.MovieRecyclerViewAdapter
+import com.everest.movieapp.api.Api
+import com.everest.movieapp.api.RetrofitHelper
+import com.everest.movieapp.databinding.FragmentCurrentYearMoviesBinding
+import com.everest.movieapp.model.MovieDb
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+
+class CurrentYearMovies : Fragment(R.layout.fragment_popular_movies) {
+
+    private lateinit var binding: FragmentCurrentYearMoviesBinding
+    private lateinit var recyclerView: RecyclerView
+    private lateinit var movieRecyclerViewAdapter: MovieRecyclerViewAdapter
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        binding = FragmentCurrentYearMoviesBinding.inflate(layoutInflater, container, false)
+        recyclerView = binding.currentYearRecyclerView
+
+        populateData()
+
+        return binding.root
+    }
+
+    private fun populateData() {
+
+        val api: Api = RetrofitHelper.getInstance().create(Api::class.java)
+        val currentYearMovieData = api.getCurrentYearMovies()
+
+        currentYearMovieData.enqueue(object : Callback<MovieDb> {
```

Comment:
This should go into repository layer. fragment should not handle this directly
Fragment should talk to ViewModel, ViewModel will talk to Repository.

Reply:
When you move this to Repository, Use sealed classes to return the response so that it can be easily handled